Sources/Workspace/Workspace+Prebuilts.swift

```diff
@@ -0,0 +1,603 @@
+                        package: prebuilt,
+                        version: packageVersion,
+                        observabilityScope: observabilityScope
+                    )
+            else {
+                continue
+            }
+
+            let hostPlatform = hostPrebuiltsPlatform
+
+            for library in prebuiltManifest.libraries {
+                for artifact in library.artifacts {
+                    guard artifact.platform == hostPlatform else {
+                        continue
+                    }
+
+                    if let path = try await prebuiltsManager
+                        .downloadPrebuilt(
+                            package: prebuilt,
+                            version: packageVersion,
+                            library: library,
+                            artifact: artifact,
+                            observabilityScope: observabilityScope
+                        )
+                    {
+                        // Add to workspace state
+                        let managedPrebuilt = ManagedPrebuilt(
+                            packageRef: prebuilt.packageRef,
+                            libraryName: library.name,
+                            path: path,
+                            products: library.products,
+                            cModules: library.cModules
+                        )
+                        self.state.prebuilts.add(managedPrebuilt)
+                        try self.state.save()
+                    }
+                }
+            }
+        }
+    }
+
+    var hostPrebuiltsPlatform: PrebuiltsManifest.Platform? {
+        if self.hostToolchain.targetTriple.isDarwin() {
+            switch self.hostToolchain.targetTriple.arch {
+            case .aarch64:
+                return .macos_aarch64
+            case .x86_64:
+                return .macos_x86_64
+            default:
+                return nil
+            }
+        } else if self.hostToolchain.targetTriple.isWindows() {
+            switch self.hostToolchain.targetTriple.arch {
+            case .aarch64:
+                return .windows_aarch64
+            case .x86_64:
+                return .windows_x86_64
+            default:
+                return nil
+            }
+        } else if self.hostToolchain.targetTriple.isLinux() {
+            // Load up the os-release file into a dictionary
+            guard let osData = try? String(contentsOfFile: "/etc/os-release")
+            else {
+                return nil
+            }
+            let osLines = osData.split(separator: "\n")
+            let osDict = osLines.reduce(into: [Substring: String]()) {
+                (dict, line) in
+                let parts = line.split(separator: "=")
+                if parts.count >= 2 {
+                    dict[parts[0]] = parts[1...].joined(separator: "=").trimmingCharacters(in: ["\""])
+                }
+            }
+
+            switch osDict["ID"] {
```

Comment:
I wonder if it's worth considering Ubuntu "compatible" distributions, since there are quite a few derivatives which are compatible. For example, For Linux Mint. So the check could be ID==ubuntu || ID_LIKE==ubuntu, and VERSION_CODENAME==x || UBUNTU_CODENAME==x

Reply:
Possibly. Things like Linux Mint isn't on our supported list but would be to enable it.

Reply:
Right, we'll never be able to support all Linux distributions, so this isn't about Mint specifically but just thinking about ways we can make this less specific to an _exact_ distro, since so many of them are derivatives of each other with fairly minimal differences.

Reply:
I need to do one more pass later through the distros as we fill out the roster for 6.1 and use the LIKE versions of the IDs where I can. Mint is an interesting use case since I've seen it both Debian-like and Ubuntu-like depending on the frontend you pick. Cinnamon for example would actually work since it claims to be ID=ubuntu but ID_LIKE=debian. More work needed there to make sure it does work where we need.

Reply:
Yeah, Mint's Wikipedia page says they have an alternate distribution which is based on Debian instead of Ubuntu -- Linux Mint Debian Edition (LMDE).

Reply:
Ah, yes, that's the one of the ones I tried. But the other actually looked like Ubuntu so should just work. But yes, I'll take a deeper look at that when I add the rest of the distros in January.